src/claim/netdata-claim.sh.in

```diff
@@ -1,451 +1,109 @@
-# HTTP Status code: 422
-# Exit code: 8
-ERROR_KEYS[8]="ErrInvalidNodeID"
-ERROR_MESSAGES[8]="invalid node id"
-
-# Error: The agent hostname is invalid; it does not fulfill the constraints
-# HTTP Status code: 422
-# Exit code: 9
-ERROR_KEYS[9]="ErrInvalidNodeName"
-ERROR_MESSAGES[9]="invalid node name"
-
-# Error: At least one of the given rooms ids is invalid; it does not fulfill the constraints
-# HTTP Status code: 422
-# Exit code: 10
-ERROR_KEYS[10]="ErrInvalidRoomID"
-ERROR_MESSAGES[10]="invalid room id"
-
-# Error: Invalid public key; the public key is empty or not present
-# HTTP Status code: 422
-# Exit code: 11
-ERROR_KEYS[11]="ErrInvalidPublicKey"
-ERROR_MESSAGES[11]="invalid public key"
+# Copyright (c) 2024 Netdata Inc.
+# SPDX-License-Identifier: GPL-3.0-or-later
 #
-# Error: Expired, missing or invalid token
-# HTTP Status code: 403
-# Exit code: 12
-ERROR_KEYS[12]="ErrForbidden"
-ERROR_MESSAGES[12]="token expired/token not found/invalid token"
-
-# Error: Duplicate agent id; an agent with the same id is already registered in the cloud
-# HTTP Status code: 409
-# Exit code: 13
-ERROR_KEYS[13]="ErrAlreadyClaimed"
-ERROR_MESSAGES[13]="already claimed"
-
-# Error: The node claiming process is still in progress.
-# HTTP Status code: 102
-# Exit code: 14
-ERROR_KEYS[14]="ErrProcessingClaim"
-ERROR_MESSAGES[14]="processing claiming"
+# %%NEW_CLAIMING_METHOD%%
 
-# Error: Internal server error. Any other unexpected error (DB problems, etc.)
-# HTTP Status code: 500
-# Exit code: 15
-ERROR_KEYS[15]="ErrInternalServerError"
-ERROR_MESSAGES[15]="Internal Server Error"
+set -e
 
-# Error: There was a timeout processing the claim.
-# HTTP Status code: 504
-# Exit code: 16
-ERROR_KEYS[16]="ErrGatewayTimeout"
-ERROR_MESSAGES[16]="Gateway Timeout"
+get_templated_value() {
+    value="$1"
+    default="$2"
+    override="$3"
 
-# Error: The service cannot handle the claiming request at this time.
-# HTTP Status code: 503
-# Exit code: 17
-ERROR_KEYS[17]="ErrServiceUnavailable"
-ERROR_MESSAGES[17]="Service Unavailable"
-
-# Exit code: 18 - Agent unique id is not generated yet.
-
-NETDATA_RUNNING=1
-
-get_config_value() {
-    conf_file="${1}"
-    section="${2}"
-    key_name="${3}"
-    if [ "${NETDATA_RUNNING}" -eq 1 ]; then
-        config_result=$(@sbindir_POST@/netdatacli 2>/dev/null read-config "$conf_file|$section|$key_name"; exit $?)
-        result="$?"
-        if [ "${result}" -ne 0 ]; then
-            echo >&2 "Unable to communicate with Netdata daemon, querying config from disk instead."
-            NETDATA_RUNNING=0
-        fi
-    fi
-    if [ "${NETDATA_RUNNING}" -eq 0 ]; then
-        config_result=$(@sbindir_POST@/netdata 2>/dev/null -W get2 "$conf_file" "$section" "$key_name" unknown_default)
+    if [ -n "${override}" ]; then
+        echo "${override}"
+    elif (echo "${value}" | grep -q '@'); then
+        echo "${default}"
+    else
+        echo "${value}"
```

Comment:
This returns an empty string when:

- no override
- `$1` is empty

Reply:
I did install from source and for some reason `@netdata_group_POST@` is an empty string.

Reply:
This should be fixed now, the original handling of the templating assumed changes that I forgot were not yet in the master branch.